src/qibolab/serialize.py

```diff
@@ -205,6 +205,35 @@ def dump_runcard(platform: Platform, path: Path):
         platform.qubits, platform.couplers
     )
 
-    path.write_text(
+    (path / RUNCARD).write_text(
         yaml.dump(settings, sort_keys=False, indent=4, default_flow_style=None)
     )
+
+
+def dump_kernels(platform: Platform, path: Path):
+    """Creates Kernels instance from platform and dumps as npz.
+
+    Args:
+        platform (qibolab.platform.Platform): The platform to be serialized.
+        path (pathlib.Path): Path that the kernels file will be saved.
+    """
+
+    # create and dump kernels
+    kernels = Kernels()
+    for qubit in platform.qubits.values():
+        if qubit.kernel is not None:
+            kernels[qubit.name] = qubit.kernel
+
+    kernels.dump(path / KERNELS)
```

Comment:
For even better isolation, eventually I'd consider having these functions just writing on byte streams, and returning them.
The top level dump could be the one responsible to attribute each byte stream a path, and dump it there.

In any case, it's not for this PR.

Reply:
The drawback of byte streams is just memory consumption: if you dump on streams, instead of directly on disk, you double the amount of memory you consume.

But I'd say CPU memory should not be an issue in Qibolab...